### DISCUSSION TITLE
Export custom properties to SVG

### DISCUSSION THREAD

**Discussion by m3-welch** [fabricjs/fabric.js#6916] [Ideas]
When using the `toJson()` function, you can specify properties to include in the export like so `canvas.toJson(['foo', 'bar'])`.

I don't see any way of doing a similar thing in the `toSVG()` function. It would make sense if it operated in the same fashion as `toJson()`.

Is this something worth opening an issue over or is it something that has already been discussed and rejected?

Thanks

**Comment by asturur** [comment]
Well SVG has standard attributes, what would be the purpose to add extra attributes that do not change rendering?

JSON is a data store while svg is a graphic file format.

**Reply by m3-welch** [reply]
Hi @asturur, thanks for the quick reply.

I am using Fabric.js to create custom SVGs for another program I use that requires extra custom attributes. I am already using the JSON export as a file store, but my software needs to produce the SVG with custom attributes as the final stage of the user flow.

I can always override the toSVG() method to perform this function, I just have seen several StackOverflow posts about the same issue, so I thought it might be worthwhile at least mentioning it on the repo.

Thanks.

**Reply by asturur** [reply]
i think that if it is a custom need and those attributes have nothing to do with the svg format, it shouldn't be part of the library. 

There is anyway a shared svg function that is shared across objects ( if i m not wrong ). if that can be changed to account for this need, we can see if the implementation is very small and in case adopt it. If it turns out to be have a bigger footprint, at least you still developed your custom solution.

**Reply by m3-welch** [reply]
That sounds brilliant.

Thanks for your help :)
